### PR TITLE
(FACT-776) Retrieve filesystem label if it exists for partitions fact, in Linux

### PR DESCRIPTION
**Pull request by vide**
Hi
this is to implement this feature request https://tickets.puppetlabs.com/browse/FACT-776 (which I opened too), I hope the implementation is OK for you. If I need to add something more, just ask (and explain how for example if it's unit testing, I'm not very good at it)

### CODE REVIEW

**Comment by vide** [comment]
CLA signed

**Comment by MikaelSmith** [comment]
This seems like a reasonable extension. Please update your commit message to follow the contributor guidelines at https://github.com/puppetlabs/facter/blob/master/CONTRIBUTING.md.

File: lib/facter/util/partitions/linux.rb [hunk 1]
```diff
@@ -55,6 +55,13 @@ def self.filesystem(partition)
       end
     end
     
+    def self.label(partition)
+      label = nil
```
Comment (MikaelSmith) [inline lib/facter/util/partitions/linux.rb]: This looks like an orphaned variable. Can it be removed?

File: lib/facter/util/partitions/openbsd.rb [hunk 2]
```diff
@@ -23,6 +23,10 @@ def self.filesystem(partition)
       scan_mount(/\/dev\/#{partition}\son\s\S+\stype\s(\S+)/)
     end
 
+    def self.label(partition)
```
Comment (MikaelSmith) [inline lib/facter/util/partitions/openbsd.rb]: Like uuid, please add a comment explaining why this is nil on OpenBSD.

**Comment by vide** [comment]
@MikaelSmith I thought about doing it but I don't have any OpenBSD installation at hand so I wasn't sure about it. Googling didn't clarify that. Anyway I will add some comment

File: lib/facter/util/partitions/linux.rb [hunk 3]
```diff
@@ -55,6 +55,13 @@ def self.filesystem(partition)
...
+      if Facter::Core::Execution.which('blkid')
+        Facter::Core::Execution.exec("blkid #{File.join('/dev', partition)}").scan(/LABEL="([^"]*)"/).flatten.first
```
Comment (MikaelSmith) [inline lib/facter/util/partitions/linux.rb]: Have you tried this with an unlabeled partition?

**Comment by MikaelSmith** [comment]
If you'd still like to try OpenBSD, http://www.vagrantbox.es/ has some Vagrant boxes that may work.

File: lib/facter/util/partitions/linux.rb (hunk 3 above)
Comment (vide) [inline lib/facter/util/partitions/linux.rb]: Yes, it doesn't return any key, so it's completely backwards compatible

File: lib/facter/util/partitions/linux.rb (hunk 1 above)
Comment (vide) [inline lib/facter/util/partitions/linux.rb]: Actually that should ensure that if there's no label everything works as
always. I can try to see if even without this assignment that scenario is
covered as wel

Comment (MikaelSmith) [inline lib/facter/util/partitions/linux.rb]: An empty function returns nil. Both #mount and #filesystem rely on that.

**Comment by vide** [comment]
I changed the path as you suggested @MikaelSmith , hope it's cool now!

**Comment by MikaelSmith** [comment]
It was just pointed out to me that we have specs that cover the partitions facts at https://github.com/puppetlabs/facter/blob/master/spec/unit/partitions_spec.rb#L21-L49. They should be extended to cover the `label`, following the same pattern used for the other facts in `partitions`.

**Comment by vide** [comment]
Ok, added the test. I put the label just in one to check if it works OK with an unlabeled filesystem too

**Comment by MikaelSmith** [comment]
Thanks for the contribution!

**Comment by vide** [comment]
Thanks! I'm quite new to gitHub so, may I ask why I don't appear as author of this commit nor contributor to facter code now?

**Comment by kylog** [comment]
@vide git can be confusing :) Every merge results in a "merge commit" whose author may (in fact, almost always is) different from the author of the individual commit(s) that make up the pull request.

So here @MikaelSmith is the author of the *merge commit* but you are the author of the code commit. To visualize this better, see https://github.com/puppetlabs/facter/commits/master where you'll see both those two commits under "Commits on Jan 15, 2015".

Hope that makes sense, and thanks again for the contribution!

**Comment by vide** [comment]
Thanks @kylog  :)

**Pull request merged**